pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -296,38 +419,52 @@ func flavorSelector(spec *corev1.PodSpec, allowedKeys sets.String) nodeaffinity.
 	return nodeaffinity.GetRequiredNodeAffinity(&corev1.Pod{Spec: specCopy})
 }
 
-// fitsFlavorLimits returns whether a requested resource fits in a specific flavor's quota limits.
+// fitsFlavorLimits returns how could this flavor be assigned, according to the
+// remaining quota in the ClusterQueue and cohort.
 // If it fits, also returns any borrowing required.
-func fitsFlavorLimits(rName corev1.ResourceName, val int64, cq *cache.ClusterQueue, flavor *cache.FlavorLimits) (int64, *Status) {
+func fitsFlavorLimits(rName corev1.ResourceName, val int64, cq *cache.ClusterQueue, flavor *cache.FlavorLimits) (FlavorAssignmentMode, int64, *Status) {
 	var status Status
 	used := cq.UsedResources[rName][flavor.Name]
+	mode := NoFit
+	if val <= flavor.Min {
+		// The request can be satisfied by the min quota, if all active workloads
+		// in the ClusterQueue are preempted.
+		mode = ClusterQueuePreempt
+	}
+	borrow := used + val - flavor.Min
+	if borrow <= 0 {
```

Comment:
It seems to me that readability could be improved by moving this `if` into the `if val <= flavor.Min {` branch to avoid mutating `mode` and to make it clear that this happens only if `val <= flavor.Min`. For example, due to the mutability of `mode` it is not obvious if the comment under `val <= flavor.Min`  describes the entering of the branch or the semantics of the mode `ClusterQueuePreempt`  - because the semantics of the mode depends on what happens later.

Reply:
I'm happy with the current flow. It might make it easier to split each mode into it's own independent function in the future, if necessary.

Reply:
I'm also happy with that approach